test/unit_tests/unit_test_image_function.cpp

```diff
@@ -351,6 +351,68 @@ namespace Function_Template
         return std::all_of( result.begin(), result.end(), [&sum]( uint32_t v ) { return v == sum; } );
     }
 
+    bool form1_BinaryDilate(BinaryDilateForm1 BinaryDilate)
+    {
+        std::vector< uint8_t > fillData( randomValue<uint32_t>(20, 200), 255u );
+        fillData.push_back(0u);
+
+        const PenguinV_Image::Image input = randomImage( fillData );
+        PenguinV_Image::Image output = input;
+
+        const uint32_t dilationX = randomValue<uint32_t>(1, 5);
+        const uint32_t dilationY = randomValue<uint32_t>(1, 5);
+
+        BinaryDilate(output, dilationX, dilationY);
+
+        return equalSize( input, output ) && verifyImage( output, 255u );
+    }
+
+    bool form2_BinaryDilate(BinaryDilateForm2 BinaryDilate)
+    {
+        std::vector <uint8_t> v (100, 255u);
+        v.push_back(0u);
+        PenguinV_Image::Image image = randomImage(v);
+
+        uint32_t roiX, roiY, roiWidth, roiHeight;
+        generateRoi ( image, roiX, roiY, roiWidth, roiHeight );
+        uint8_t result = !verifyImage (image, roiX, roiY, roiWidth, roiHeight, 0u) ? 255u : 0u;
```

Comment:
Could you please explain why do we need to do this?

Reply:
verifyImage will return white if the random image region can be dilated, and if the region is fully black (and can't be dilated), black will be returned.  I was comparing this to the value returned from BinaryDilate to see if the dilation results matched what I predicted.  If it's not right, let me know why, or if there's an easier or better way, let me know.

I'll pull that makefile out and put it in a new PR.

Reply:
Okay. I got the idea. I corrected the logic of functions to be more obvious for other people.